scripts/language_model/word_language_model.py

```diff
@@ -354,30 +357,51 @@ def train():
             data_list = gluon.utils.split_and_load(data, context, batch_axis=1, even_split=True)
             target_list = gluon.utils.split_and_load(target, context, batch_axis=1, even_split=True)
             hiddens = detach(hiddens)
+            model.collect_params().zero_grad()
```

Comment:
Why do you zero out the gradient? I think you are not using accumulated gradient, right?

Reply:
Based on the original implementation of the paper, https://github.com/salesforce/awd-lstm-lm/blob/32fcb42562aeb5c7e6c9dec3f2a3baaaf68a5cb5/main.py#L194 , it is used before updating parameters.

Reply:
They use torch, in which the gradient is accumulated across the iterations in default. In gluon, the gradient is replaced in default.